src/Shape.cpp

```diff
@@ -100,11 +100,57 @@ double line_parameter(const Vector3 &D, const Vector3 &v1, const Vector3 &v2, co
     return p_v1 + (p_v2-p_v1) * d_v1/(d_v1-d_v2);
 }
 
+double crossProduct(const Vec2d& v1, const Vec2d& v2) {
+    return v1.x*v2.y - v1.y*v2.x;
+}
+
+bool linelineintersect(const Vec2d& v1, const Vec2d& v2, const Vec2d& w1, const Vec2d& w2) {
+    Vec2d ev = v2-v1;
+    Vec2d ew = w2-w1;
+    // check for parallel lines
+    double ev_x_ew = crossProduct(ev, ew);
+    if (ev_x_ew == 0) // lines are parallel
+    {
```

Comment:
Be consistent with brackets, following the standard of the rest of the library. So opening bracket on same line.